thrift-parser.js

```diff
@@ -366,6 +410,34 @@ module.exports = (buffer, offset = 0) => {
     return result;
   };
 
+  const readUnion = () => {
+    let subject = readKeyword('union');
+    let name = readName();
+    let items = readUnionBlock();
```

Comment:
`readUnionBlock` was needed because it must call `readUnionItem` so it can drop any `option` keyword, which is different behavior than `readStructItem` (see https://github.com/eleme/thrift-parser/pull/6/files#diff-678157f157421c8cb7becf319d0e3b99R431).  I thought about re-using `readStructBlock` but the code would have had to iterate over the `items` and delete each `option` property, which seemed worse than having separated logic.